client/repo.c

```diff
@@ -693,6 +694,56 @@ TDNFGetRepoMD(
     BAIL_ON_TDNF_ERROR(dwError);
     *ppRepoMD = pRepoMD;
 
+    if (pRepoData->pszSnapshotUrl) {
+        time_t now = time(NULL);
+        int needDownload = 0, nIsRemote = 0;
+        struct stat st = {0};
+
+        dwError = TDNFUriIsRemote(pRepoData->pszSnapshotUrl, &nIsRemote);
```

Comment:
Check if it is local file first and then see if it is valid url and there should be a error message thrown accordingly.